### PR TITLE
make google-apps-script up-to-date, with some fixes

### PR DESCRIPTION
**Pull request by motemen**
Updated declarations up-to-date with the document, generated by https://github.com/motemen/dts-google-apps-script.

Also, fixed some issues:

- `Service.Restriction` was incorrectly named `Service`.
- Enum fields had wrong types.

----

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://developers.google.com/apps-script/
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dslint/dt.json" }`.

### CODE REVIEW

File: types/google-apps-script/google-apps-script.calendar.d.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
-// Type definitions for Google Apps Script 2015-11-12
+// Type definitions for Google Apps Script 2017-05-12
 // Project: https://developers.google.com/apps-script/
 // Definitions by: motemen <https://github.com/motemen/>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-
-
+/// <reference path="google-apps-script.types.d.ts" />
```
Comment (aozgaa) [inline types/google-apps-script/google-apps-script.calendar.d.ts]: Should this be `// < reference type=""/>`?

File: types/google-apps-script/google-apps-script.cache.d.ts [hunk 2]
```diff
@@ -1,9 +1,9 @@
-// Type definitions for Google Apps Script 2015-11-12
+// Type definitions for Google Apps Script 2017-05-12
 // Project: https://developers.google.com/apps-script/
 // Definitions by: motemen <https://github.com/motemen/>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-
+/// <reference path="google-apps-script.types.d.ts" />
```
Comment (aozgaa) [inline types/google-apps-script/google-apps-script.cache.d.ts]: ditto

File: types/google-apps-script/google-apps-script.charts.d.ts [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
+// Type definitions for Google Apps Script 2017-05-12
 // Project: https://developers.google.com/apps-script/
 // Definitions by: motemen <https://github.com/motemen/>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-
-
-
+/// <reference path="google-apps-script.types.d.ts" />
+/// <reference path="google-apps-script.base.d.ts" />
```
Comment (aozgaa) [inline types/google-apps-script/google-apps-script.charts.d.ts]: ditto

File: types/google-apps-script/google-apps-script.contacts.d.ts [hunk 4]
```diff
@@ -1,10 +1,10 @@
-// Type definitions for Google Apps Script 2015-11-12
+// Type definitions for Google Apps Script 2017-05-12
 // Project: https://developers.google.com/apps-script/
 // Definitions by: motemen <https://github.com/motemen/>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-
-
+/// <reference path="google-apps-script.types.d.ts" />
```
Comment (aozgaa) [inline types/google-apps-script/google-apps-script.contacts.d.ts]: ditto

File: types/google-apps-script/google-apps-script-tests.ts [hunk 5]
```diff
@@ -25,3 +25,9 @@ function createAndSendDocument() {
   // Send yourself an email with a link to the document.
   GmailApp.sendEmail(email, subject, body);
 }
+
+// Regression
```
Comment (aozgaa) [inline types/google-apps-script/google-apps-script-tests.ts]: What is the regression?

File: types/google-apps-script/google-apps-script.calendar.d.ts (hunk 1 above)
Comment (motemen) [inline types/google-apps-script/google-apps-script.calendar.d.ts]: Thanks, I didn't know [`<reference types=...>` directive](https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html#-reference-types-) and have just read the doc.

Are you saying this line should be `/// <reference types="google-apps-script" />`? I think it causes kind of recursive reference and does not achieve what I wanted to do; I just wanted to import type declarations such as `GoogleAppsScript.Integer` which is declared in google-apps-script.types.d.ts.

Maybe whole these references could be deleted, as index.d.ts references all files.

File: types/google-apps-script/google-apps-script-tests.ts (hunk 5 above)
Comment (motemen) [inline types/google-apps-script/google-apps-script-tests.ts]: Ah, I meant the bugs I wrote in the PR body:

> - `Service.Restriction` was incorrectly named `Service`.
> - Enum fields had wrong types.

Should I write comments more precisely?

**Pull request merged**